sunpy/net/dataretriever/sources/goes.py

```diff
@@ -91,19 +91,19 @@ def _get_goes_sat_num(self, date):
     def _get_time_for_url(self, urls):
         times = []
         for uri in urls:
-            uripath = urlsplit(uri).path
-
-            # Extract the yymmdd or yyyymmdd timestamp
-            datestamp = os.path.splitext(os.path.split(uripath)[1])[0][4:]
-
-            # 1999-01-15 as an integer.
-            if int(datestamp) <= 990115:
-                start = Time.strptime(datestamp, "%y%m%d")
+            if uri.endswith('fits'):
+                datestamp = os.path.basename(uri)[4:12]
```

Comment:
If it ends with `fits`, then seems like it doesn't check whether the pattern is `%Y%m%d` or `%y%m%d`, i.e., that check around the date `1999/1/15`.

Reply:
good catch - thanks